### ISSUE TITLE
RGUI Onscreen Notifications disappear

### ISSUE THREAD

**Issue by RiskyJumps** [libretro/RetroArch#6607]
## Description

When Display Framerate is ON, onscreen notifications --like shader names and save/load states messages-- appear for a single frame and then they get overwritten by the framerate notification.

### Expected behavior

New notifications are displayed for some seconds in place of the framerate message, then they disappear and framerate message displays instead.

### Actual behavior

New notifications are displayed and disappear after one frame as the framerate message gets spammed over it every frame.

### Steps to reproduce the bug

1. Go to `Settings > Onscreen Display > Onscreen Notifications ` and turn it `ON`
2. Go to `Settings > Onscreen Display > Display Framerate ` and turn it `ON`
3. Switch shaders using hotkeys or load/save a state.

### Version/Commit
- RetroArch: 1.7.2
- Build date: Apr 22 2018
- Git version: 997c24a

### Environment information

- OS: Ubuntu 16.04 LTS 64-bit. Kernel: 4.10.0-40-generic.
- Compiler: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609.

**Comment by Laynee** [comment]
I have the same problem, it's very annoying. I'm on Windows 10 + Nvidia GTX 1050. It's not specific to RatroArch 1.7.2. It was the same on 1.7.0 and maybe before.

In my case, the problème seems related to the graphic card / driver. With an ATI integrated card, notifications were working fine. As soon as I updated to the Nvidia card, they became 1 frame only / invisible. Updating drivers does nothing.

What is your graphic card?

**Comment by RiskyJumps** [comment]
That's right. I remember it started some time ago. My graphic cards are both Nvidia, so you might be right on it too.

**Comment by RiskyJumps** [comment]
I found it. The commit that started the issue was [5a990841dc0753b6a3a218d345dad72ea6cf424b](https://github.com/libretro/RetroArch/commit/5a990841dc0753b6a3a218d345dad72ea6cf424b).

**Comment by RiskyJumps** [comment]
If its acceptable, here is a working [patch](https://pastebin.com/4wLY3uHk).

**Comment by Laynee** [comment]
Nice find ! So it was just this line in video_driver.c that broke notifications on Nvidia cards when fps_show  is on :

```
-      runloop_msg_queue_push(video_info.fps_text, 1, 1, false);
+      runloop_msg_queue_push(video_info.fps_text, 2, 1, true);
```

@RobLoach Can you look into it ?

**Comment by RiskyJumps** [comment]
@Laynee That's right. But, it pains me to say this, the patch won't leave it working just as it was before the issue. After you switch shaders using hotkeys or change savestates slot you won't get load/save messages. You still get the other messages, though. It still needs further examination.

**Comment by keithbowes** [comment]
> I found it. The commit that started the issue was [5a990841dc0753b6a3a218d345dad72ea6cf424b](https://github.com/libretro/RetroArch/commit/5a990841dc0753b6a3a218d345dad72ea6cf424b).

That should be easy to fix.  I'd suggest a new `fps_has_priority` setting (with the UI being "Always show FPS instead of other messages" (perhaps as a sublabel of something more concise)).
